### PR TITLE
Add validation for child jobs without ownerReference

### PR DESCRIPTION
**Pull request by tenzen-y**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
I added validation not to allow child jobs without ownerReference.

#### Which issue(s) this PR fixes:

Follow-up: https://github.com/kubernetes-sigs/kueue/pull/821#discussion_r1211608721

#### Does this PR introduce a user-facing change?

```release-note
Add validation for child jobs without ownerReference.
```

### CODE REVIEW

**Comment by tenzen-y** [comment]
friendly ping @kerthcet :)

File: pkg/controller/jobframework/validation.go [hunk 1]
```diff
@@ -57,6 +57,16 @@ func validateLabelAsCRDName(job GenericJob, crdNameLabel string) field.ErrorList
 	return allErrs
 }
 
+func ValidateCreateForParentWorkload(job GenericJob) field.ErrorList {
+	var allErrs field.ErrorList
+	if value, exists := job.Object().GetAnnotations()[constants.ParentWorkloadAnnotation]; exists {
+		if job.Object().GetOwnerReferences() == nil {
+			allErrs = append(allErrs, field.Invalid(parentWorkloadKeyPath, value, "must not add a parent workload annotation to job without OwnerReference"))
```
Comment (kerthcet) [inline pkg/controller/jobframework/validation.go:64]: I think we should use field.Forbidden.

Comment (tenzen-y) [reply pkg/controller/jobframework/validation.go:64]: It makes sense.

Comment (tenzen-y) [reply pkg/controller/jobframework/validation.go]: Done.

**Comment by tenzen-y** [comment]
@kerthcet I have addressed your comments. PTAL.

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve
/hold
for suggestion, but it can also be a follow up

**Pull request merged**